### DISCUSSION TITLE
creating a dynamic table in the js(go) templating engine

### DISCUSSION THREAD

**Discussion by gokulgseven** [pocketbase/pocketbase#4573] [Q&A] [answered]
i am trying to create a simple table that will display the columns and the corresponding data 
`tableColumns = [{name:"username", type: "text" }, {name:"email", type: "email"}, {name: id, type: "text"}]`
this is a list of objects describing the column headings

`data = [{username: "gokul", email: "em@gm.com", id: "001"},{username: "abh", email: "abh@gm.com", extrafield: "this will not display"} ]`
this is a list of objects containing the data

```
<table>
        <thead>
            <tr>
                {{range $index, $col := .tableColumns}}
                <th>{{ $col.name }}</th>
                {{end}}
            </tr>
        </thead>
        <tbody>
            {{range $i, $row := .data}}
            <tr>
                {{range $index, $col := $.tableColumns}}
                <td>{{$row.GetFieldValue $col.name}}</td> 
                {{end}}
            </tr>
            {{end}}
        </tbody>
    </table>
```
i have tried multiple combinations of things to get this to work , but as soon as i nest two ranges inside one another everything stops working 

this doesnt work 
`  <td>{{$row[$col.name]}}</td> `

**Comment by ganigeorgiev** [comment, marked as answer]
You can try using the [`index` template function](https://pkg.go.dev/text/template#hdr-Functions), something like:

```html
 <td>{{index $row $col.name}}</td>
```
